app/models/event.rb

```diff
@@ -1,4 +1,6 @@
 class Event < ApplicationRecord
+  before_update :destroy_registrations, if: :cancelled_changed?
```

Comment:
Shouldn't this be after_update? What if it destroys registrations and we get error in changing cancelled boolean.

Actually, it should be ```after_commit on update```, please read and let me know why I am saying this